### DISCUSSION TITLE
Repeatedly gives the same suggestion even after accepting it

### DISCUSSION THREAD

**Discussion by snipeart007** [community/community#7896] [Copilot Conversations] [answered]
I am learning DSA using C++. But when I accept a suggestion given on the basis of my class name and function name, it repeatedly gives the same suggestion. I was implementing my ```DataStructures::Queue<Type>::dequeue```. When I hit enter from my opening bracket it again suggests the whole function. When I accept it, the body of the function is doubled.

```cpp
Queue<Type>::dequeue()
{
    if (isEmpty() == true)
    {
        std::cout << "Queue is Empty";
    }

    else
    {
        Type element = this->arr[front];
        this->front++;
        return element;
    }
    if (isEmpty() == true)
    {
        std::cout << "Queue is Empty";
    }

    else
    {
        Type element = this->arr[front];
        this->front++;
        return element;
    }
}
```

Here's a screenshot.
![image](https://user-images.githubusercontent.com/84833041/142579490-cfa6593c-ab90-4112-a336-4cb7ad1e4e43.png)

**Comment by MulleDK19** [comment, marked as answer]
This is expected. It only knows about the stuff from the beginning of the file to the current caret position.
Hit escape to dismiss the suggestion.

**Reply by snipeart007** [reply]
Thanks. Didn't know about the fact that it only takes data from the beginning till the current caret position.
